OsmAnd/src/net/osmand/plus/routing/TransportRoutingHelper.java

```diff
@@ -580,7 +586,12 @@ public void onCalculationFinish() {
 						walkingSegmentsCalculated = true;
 					} else {
 						onUpdateCalculationProgress(0);
-						RouteCalculationParams walkingRouteParams = getWalkingRouteParams();
+						RouteCalculationParams walkingRouteParams;
+						do {
+							walkingRouteParams = getWalkingRouteParams();
+							walkingSegmentsCalculated = walkingSegmentsToCalculate.isEmpty();
+						}
+						while (walkingRouteParams == null && !walkingSegmentsCalculated);
```

Comment:
This should be reverted. Modify and rename `getWalkingRouteParams` to `getOrProcessWalkingRouteParams`